tc_lio_sam:
  # Topics
  pointCloudTopic: "os_cloud_node/points"               # Point cloud data
  imuTopic: "os_cloud_node/imu"                         # IMU data
  odomTopic: "odometry/imu"                   # IMU pre-preintegration odometry, same frequency as IMU
  # Frames
  lidarFrame: "base_link"
  baseFrame: "base_link"
  odomFrame: "odom"
  mapFrame: "map"
  # Export settings
  savePCD: false                              # https://github.com/TixiaoShan/LIO-SAM/issues/3
  savePCDDirectory: "/Downloads/LOAM/"        # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "MapOptimization" for implementation
  # Sensor Settings
  sensor: ouster                            # lidar sensor type, 'velodyne' or 'ouster' or 'livox'
  N_SCAN: 128                                  # number of lidar channel (i.e., Velodyne/Ouster: 16, 32, 64, 128, Livox Horizon: 6)
  Horizon_SCAN: 1024                          # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048, Livox Horizon: 4000)
  downSampleRate: 1                           # default: 1. Downsample your data if too many points. i.e., 16 = 64 / 4, 16 = 16 / 1
  lidarMinRange: 1.0                          # default: 1.0, minimum lidar range to be used
  lidarMaxRange: 1000.0                       # default: 1000.0, maximum lidar range to be used
  # IMU Settings
  imuAccNoise: 0.1
  imuGyrNoise: 0.1
  imuAccBiasN: 0.0001
  imuGyrBiasN: 0.0001
  imuGravity: 9.80511
  imuRpyWeight: 0.01
  # Extrinsics (lidar -> IMU)
  extrinsicTrans: [0.014, -0.012, -0.015]
  extrinsicRot: [1, 0, 0,
                 0, 1, 0,
                 0, 0, 1]
  extrinsicRPY: [1, 0, 0,
                 0, 1, 0,
                 0, 0, 1]
  # LOAM feature threshold
  edgeThreshold: 1.0
  surfThreshold: 0.1
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100
  # voxel filter paprams
  odometrySurfLeafSize: 0.4                     # default: 0.4 - outdoor, 0.2 - indoor
  mappingCornerLeafSize: 0.2                    # default: 0.2 - outdoor, 0.1 - indoor
  mappingSurfLeafSize: 0.4                      # default: 0.4 - outdoor, 0.2 - indoor
  # CPU Params
  numberOfCores: 4                              # number of cores for mapping optimization
  mappingProcessInterval: 0.15                  # seconds, regulate mapping frequency
  # Surrounding map
  surroundingKeyFrameAddingDistThreshold: 1.0   # meters, regulate keyframe adding threshold
  surroundingKeyFrameAddingAngleThreshold: 0.2  # radians, regulate keyframe adding threshold
  surroundingKeyFrameDensity: 2.0               # meters, downsample surrounding keyframe poses
  surroundingKeyFrameSearchRadius: 50.0         # meters, within n meters scan-to-map optimization (when loop closure disabled)
  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
  surroundingKeyFrameSize: 50                   # submap size (when loop closure enabled)
  historyKeyFrameSearchRadius: 15.0             # meters, key frame that is within n meters from current pose will be considerd for loop closure
  historyKeyFrameSearchTimeDiff: 30.0           # seconds, key frame that is n seconds older will be considered for loop closure
  historyKeyFrameSearchNum: 25                  # number of hostory key frames will be fused into a submap for loop closure
  historyKeyFrameFitnessScore: 0.3              # icp threshold, the smaller the better alignment
  # Visualization
  globalMapVisualizationSearchRadius: 1000.0    # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 1.0           # meters, global map visualization cloud density